hack/update-manifests.sh

```diff
@@ -27,23 +27,34 @@ if [ "$IMAGE_TAG" = "" ]; then
   IMAGE_TAG=latest
 fi
 
+# bundle_with_addons bundles given kustomize base with either stable or latest version of addons
+function bundle_with_addons() {
+  for addon in $(ls $SRCROOT/manifests/addons | grep -v README.md); do
+    ADDON_BASE="stable"
+    if [ "$IMAGE_TAG" = "latest" ]; then
+        ADDON_BASE="latest"
+    fi
```

Comment:
I'm not sure whether testing for `$IMAGE_TAG" = "latest"` will produce desired results. Also, this will be the source of some weird side-effects of using `IMAGE_TAG` (which should only set the image tag imho).

I can't come up quickly with a better solution for now, but maybe we should re-evaluate this in the future.

Reply:
Good catch. I was trying to re-use this check: https://github.com/argoproj/argo-cd/blob/98bec43aaa3b86311473ba24e5302b3da41fe59b/hack/update-manifests.sh#L20 to avoid running `git rev-parse` second time but did not realise that `IMAGE_TAG` env variables can be specified by user. Switched to just using `git rev-parse` directly